arroyo/processing/strategies/unfold.py

```diff
@@ -48,18 +48,13 @@ def submit(self, message: Message[Union[FilteredPayload, TInput]]) -> None:
             return
 
         iterable = self.__generator(message.payload)
-        num_messages = len(iterable)
 
         store_remaining_messages = False
 
-        for idx, value in enumerate(iterable):
-            # Last message is special because offsets can be committed along with it
-            if idx == num_messages - 1:
-                next_message = Message(
-                    Value(value, message.committable, message.timestamp)
-                )
-            else:
-                next_message = Message(Value(value, {}, message.timestamp))
+        for value in iterable:
+            next_message = Message(
+                value=cast(Value[TOutput], value),
```

Comment:
I don't think this cast should be necessary, more likely somewhere else the types weren't updated